src/models/SnackbarStore.ts

```diff
@@ -0,0 +1,70 @@
+import { SnapshotOrInstance, types, cast, Instance, flow, getRoot, SnapshotIn } from 'mobx-state-tree'
+import { autorun } from 'mobx'
+import { createId } from '@/utils/createId'
+
+export type SnackbarIconType = 'success' | 'error' | 'info' | 'warning'
+
+const SNACKBARS_LIMIT = 3
+
+export const Snackbar = types
+  .model('Snackbar', {
+    id: types.optional(types.identifier, () => createId()),
+    title: types.string,
+    description: types.maybe(types.string),
+    timeout: types.maybe(types.number),
+    variant: types.maybe(types.enumeration(['primary', 'secondary'])),
+    iconType: types.maybe(types.enumeration(['success', 'error', 'info', 'warning'])),
+    actionText: types.maybe(types.string),
+  })
+  .volatile((self) => ({
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    onActionClick() {},
+  }))
```

Comment:
What will happen if we don't add this?

Reply:
that adds some temp state to the Model and makes it aware of it so later on we can set a custom function from outside with the `setOnActionClick` action